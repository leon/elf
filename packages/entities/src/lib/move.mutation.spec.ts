import {
  createEntitiesStore,
  createTodo,
  toMatchSnapshot,
} from '@ngneat/elf-mocks';
import { addEntities } from './add.mutation';
import { moveEntity } from './move.mutation';

describe('move', () => {
  let store: ReturnType<typeof createEntitiesStore>;

  beforeEach(() => {
    store = createEntitiesStore();
  });

  it('should move entity', () => {
    store.update(addEntities([createTodo(1), createTodo(2)]));
    store.update(moveEntity({ fromIndex: 0, toIndex: 1 }));
    toMatchSnapshot(expect, store, 'move 1 to 2');
  });
});
