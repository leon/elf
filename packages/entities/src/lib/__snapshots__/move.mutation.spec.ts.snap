// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`move should move entity: move 1 to 2 1`] = `
Object {
  "entities": Object {
    "1": Object {
      "completed": false,
      "id": 1,
      "title": "todo 1",
    },
    "2": Object {
      "completed": false,
      "id": 2,
      "title": "todo 2",
    },
  },
  "ids": Array [
    2,
    1,
  ],
}
`;
